### DISCUSSION TITLE
Issues using the API - response using api differs vastly from response in web interface

### DISCUSSION THREAD

**Discussion by BigKid1973** [oobabooga/text-generation-webui#3631] [Q&A]
After a long period of try and error I finally got the api to respond to my c# client.

However the answer is very different from the answer I get using the web-interface.
I use "instruct" mode in web interface and api.

In the webinterface in the setting for chat-instruct I modified Context = "" and Command for chat-instruct mode = "<|prompt|>".

The answers I get using the API remind me of the answers I got before that.Like the LLM is reflecting ABOUT my request instead of fullfilling it.
When trying very often I sometimes get an answer alike what I want and usually get on the webinterface but again with some kind of introductory sentence like the AI is reflecting ABOUT my request first instead of fullfilling it.

Any help highly appreciated. I'm just looking for a way to get the same response I would get when typing it into theh web interface using the same settings I made there (or just use those settings). Probably I'm using the wrong api, or I use it wrong but all I found are the examples here.

This is the json I send to http://127.0.0.1:5001/v1/generate_text

```                   
string json = JsonSerializer.Serialize(new {
                        prompt = sChatMessage + " ",
                        auto_max_new_tokens = false,
                        max_new_tokens = 4096,
                        history = "",
                        mode = "instruct",
                        character = "None",
                        your_name = "You",
                        context = "",
                        context_instruct = "",
                        instruction_template = "Vicuna-v1.1",
                        preset = "simple-1",
                        regenerate = false,
                        _continue = false,
                        chat_instruct_command = "<|prompt|> ",
                        instruct_command = "<|prompt|> ",
                        do_sample = true,
                        temperature= 0.7,
                        top_p= 0.9,
                        typical_p= 1,
                        repetition_penalty= 1.15,
                        encoder_repetition_penalty= 1.0,
                        top_k= 20,
                        min_length= 0,
                        no_repeat_ngram_size= 0,
                        num_beams= 1,
                        penalty_alpha= 0,
                        length_penalty= 1,
                        early_stopping= false,
                        seed= -1,
                        add_bos_token= true,
                        truncation_length= 8196,
                        ban_eos_token= false,
                        skip_special_tokens= true
}
```

**Comment by klaasvk** [comment]
Same shit lil bro

**Comment by Mikec78660** [comment]
I see the same thing when I use the API vs the webui. I'm not sure what was causing it. I thought I had all the settings identical. Mine was adding things at the end. Like asking 3 times if I wanted more information and stuff like that.
